Panic during replay-live

I can reliably reproduce a panic `close of closed channel` when performing replay-live actions.

```
panic: close of closed channel

goroutine 201 [running]:
panic(0xb82b60, 0xc820bea0a0)
        /usr/lib/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/kapacitor.(*Edge).Close(0xc820e676c0)
        /home/nathanielc/projects/kapacitor/src/github.com/influxdata/kapacitor/edge.go:121 +0x1c6
github.com/influxdata/kapacitor/services/replay.(*Service).doReplay.func1(0xc820a27380, 0xc820213a40)
        /home/nathanielc/projects/kapacitor/src/github.com/influxdata/kapacitor/services/replay/service.go:1186 +0x124
github.com/influxdata/kapacitor/services/replay.(*Service).doReplay(0xc8200eb760, 0xc820a753e0, 0x24, 0xc820213a40, 0xc820a8e720, 0x7f67513e88b0, 0xc820e62930)
        /home/nathanielc/projects/kapacitor/src/github.com/influxdata/kapacitor/services/replay/service.go:1194 +0x29b
github.com/influxdata/kapacitor/services/replay.(*Service).doLiveBatchReplay(0xc8200eb760, 0xc820a753e0, 0x24, 0xc820213a40, 0x7f6751470058, 0xc820a7b860, 0xc820cda001, 0xecdaf285c, 0x27c61c74, 0x14962a0, ...)
        /home/nathanielc/projects/kapacitor/src/github.com/influxdata/kapacitor/services/replay/service.go:1129 +0x136
github.com/influxdata/kapacitor/services/replay.(*Service).handleReplayBatch.func1(0xc8200eb760, 0xc820ae98c0, 0xc820213a40, 0x7f6751470058, 0xc820a7b860, 0xc820a753e0, 0x24, 0x0, 0x0, 0xc820be9740, ...)
        /home/nathanielc/projects/kapacitor/src/github.com/influxdata/kapacitor/services/replay/service.go:1006 +0xb9
created by github.com/influxdata/kapacitor/services/replay.(*Service).handleReplayBatch
        /home/nathanielc/projects/kapacitor/src/github.com/influxdata/kapacitor/services/replay/service.go:1008 +0x8cd
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -30,6 +30,8 @@ stream
 
 ### Bugfixes
 
+- [#1045](https://github.com/influxdata/kapacitor/issues/1045): Fix panic during replays.
+
 
 
 ## v1.1.0 [2016-10-07]

--- a/edge.go
+++ b/edge.go
@@ -31,6 +31,9 @@ type BatchCollector interface {
 }
 
 type Edge struct {
+	mu     sync.Mutex
+	closed bool
+
 	stream chan models.Point
 	batch  chan models.Batch
 
@@ -108,7 +111,14 @@ func (e *Edge) readGroupStats(f func(group models.GroupID, collected, emitted in
 
 // Close the edge, this can only be called after all
 // collect calls to the edge have finished.
+// Can be called multiple times.
 func (e *Edge) Close() {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	if e.closed {
+		return
+	}
+	e.closed = true
 	e.logger.Printf(
 		"D! closing c: %d e: %d\n",
 		e.collected.IntValue(),

